src/sources/kubernetes_logs/parser/cri.rs

```diff
@@ -27,86 +29,144 @@ pub const NEW_LINE_TAG: &str = "new_line_tag";
 #[derivative(Debug)]
 pub(super) struct Cri {
     #[derivative(Debug = "ignore")]
-    regex_parser: Box<dyn FunctionTransform>,
+    pattern: Regex,
+    capture_locations: CaptureLocations,
+    capture_names: Vec<(usize, String)>,
+    field: &'static str,
 }
 
-impl Cri {
-    /// Create a new [`Cri`] parser.
-    pub fn new(timezone: TimeZone) -> Self {
-        let regex_parser = {
-            let mut rp_config = RegexParserConfig::default();
+impl Default for Cri {
+    fn default() -> Self {
+        let pattern =
+            Regex::new(CRI_REGEX_PATTERN).expect("CRI log regex pattern should never fail");
 
-            let pattern = r"(?-u)^(?P<timestamp>.*) (?P<stream>(stdout|stderr)) (?P<multiline_tag>(P|F)) (?P<message>.*)(?P<new_line_tag>\n?)$";
-            rp_config.patterns = vec![pattern.to_owned()];
+        let capture_names = pattern
+            .capture_names()
+            .enumerate()
+            .filter_map(|(i, s)| s.map(|s| (i, s.to_string())))
+            .collect::<Vec<_>>();
+        let capture_locations = pattern.capture_locations();
 
-            rp_config.types.insert(
-                crate::config::log_schema().timestamp_key().to_string(),
-                "timestamp|%+".to_owned(),
-            );
-
-            let parser = RegexParser::build(&rp_config, timezone)
-                .expect("regexp patterns are static, should never fail");
-            parser.into_function()
-        };
-
-        Self { regex_parser }
+        Self {
+            pattern,
+            capture_locations,
+            capture_names,
+            field: crate::config::log_schema().message_key(),
+        }
     }
 }
 
 impl FunctionTransform for Cri {
-    fn transform(&mut self, output: &mut OutputBuffer, event: Event) {
-        let mut buf = OutputBuffer::with_capacity(1);
-        self.regex_parser.transform(&mut buf, event);
-        if let Some(mut event) = buf.into_events().next() {
-            if normalize_event(event.as_mut_log()).ok().is_some() {
-                output.push(event);
+    fn transform(&mut self, output: &mut OutputBuffer, mut event: Event) {
+        // Get the log field with the message, if it exists, and coerce it to bytes.
+        let log = event.as_mut_log();
+        let value = log.get(self.field).map(|s| s.coerce_to_bytes());
+        match value {
+            None => {
+                // The message field was missing, inexplicably.
+                //
+                // This is definitely wrong so make sure we track this. We'll still ultimately
+                // end up passing the event along, though.
+                emit!(ParserMissingFieldError { field: self.field });
+                return;
```

Comment:
Do we actually? This looks like it returns without pushing the even into `output`.

Reply:
You're right: we're not actually emitting the event in this case. I'll remove that comment.

Reply:
Addressed in b1ec38102.